src/scry/symbol.cr

```diff
@@ -115,4 +116,26 @@ module Scry
       ResponseMessage.new(@text_document.id, visitor.symbols)
     end
   end
+
+  class WorkspaceSymbolProcessor
+    def initialize(@msg_id : Int32 | String, @root_path : String, @query : String)
+      @all_files = Dir.glob("#{root_path}/**/*.cr")
+    end
+
+    def run
+      symbols = [] of SymbolInformation
+      unless @query.empty?
+        @all_files.each do |file|
+          visitor = SymbolVisitor.new("file://#{file}")
+          parser = Crystal::Parser.new(File.read(file))
+          parser.filename = file
+          node = parser.parse
```

Comment:
what if there is a syntax error in one of the file? I think the exception needs to be rescued and the file skipped.

Is it possible to send errors in addition to valid results? (say, here are the results for files X & Y, but there is an error with file Z)

Reply:
I guess I can handle the error, but the response should still be managed by diagnostic provider

I think I can return an empty array if syntax error is found

Reply:
Looking at https://microsoft.github.io/language-server-protocol/specification#workspace-symbols-request-leftwards_arrow_with_hook you should return null and an error in case an exception happens during the workspace symbol request.

I think it could be possible to return a result & an error? but how the valid_result/error is handled would vastly depend on the language client implementation..

Reply:
Well, Actually we're managing that via `response_error.cr`, although I think for better user experience we should ignore parsing errors here since the diagnostics already do this work.

I don't want to see a popup saying "LSP server has an error" every time I try to list symbols and I have files with syntax errors :sweat_smile: